### PR TITLE
Rework transactional letters

### PR DESCRIPTION
**Pull request by radiansz**

### CODE REVIEW

File: src/resources/transactionalDraft/collection.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        model: Model,
+        action: 'issue.draft.list',
+
+        parse: function() {
+            var self = this;
+            var data = Collection.prototype.parse.apply(this, arguments);
+
+            var filter = _.filter(data, function(elem) {
+                var elemData = elem.obj;
+                if (self.model.prototype.isDraftId(elemData.id) || self.model.prototype._codeToType(elemData.relref) !== 'unknown') {
```
Comment (zouxuoz) [inline src/resources/transactionalDraft/collection.js]: why use private method?

File: src/resources/transactionalDraft/model.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                reltype: -1000,
+                relref: this._typeToCode(this.get('type'))
+            });
+            result['update.date'] = moment(new Date()).format('YYYY-MM-DD HH:mm:ss');
+            
+            result.letter = {};
+            result.letter['from.name'] = this.get('fromName');
+            result.letter['from.email'] = this.get('fromEmail');
+            result.letter['subject'] = this.get('subject');;
+            if(this.get('presentation')) {
```
Comment (zouxuoz) [inline src/resources/transactionalDraft/model.js]: whitespace after `if`

**Comment by zouxuoz** [comment]
missed validation for letter name on first step of wizard

**Comment by radiansz** [comment]
Validation's been added

File: src/blocks/table/letters/letters.js [hunk 3]
```diff
@@ -8,6 +8,10 @@ define(function(require) {
         entityName: 'письмо',
         deleteNotificationText: function(number) {
             return i18n('Отмеченное письмо удалено', number);
+        },
+
+        _buildUrl: function(letter) {
```
Comment (rosendi) [inline src/blocks/table/letters/letters.js]: Rename to `_constructUrl`

File: src/blocks/wizard/__step/letterEditContent/letterEditContent.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        initialize: function() {
+        	var _super = Step.prototype.initialize.apply(this, arguments);
+
+        	this.modifiers.hideActions = this.parent.modifiers.mode === 'create';
+
+        	return _super;
+        },
+
+        activate: function() {
+        	var _super = Step.prototype.activate.apply(this, arguments);
```
Comment (rosendi) [inline src/blocks/wizard/__step/letterEditContent/letterEditContent.js]: !

File: src/blocks/wizard/__step/letterSave/letterSave.js [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
     var Step = require('blocks/wizard/__step/__step');
+    var app = require('app');
+    var template = require('tpl!./letterSave.ejs'); 
 
     return Step.extend({
-        template: require('tpl!./letterSave.ejs')
+        template: template,
+
+        _save: function() {
+            var self = this;
```
Comment (rosendi) [inline src/blocks/wizard/__step/letterSave/letterSave.js]: Add new line

File: src/blocks/wizard/__step/templateEditContent/templateEditContent.js [hunk 6]
```diff
@@ -26,9 +26,9 @@ define(function(require) {
         getState: function() {
             if (!this.models.template.has('presentation')) {
                 return 'create';
```
Comment (rosendi) [inline src/blocks/wizard/__step/templateEditContent/templateEditContent.js]: `undone`

File: src/pages/automation/letters/content.ejs [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 <div class="content content_tightened">
     <%= block.include(require('blocks/table/letters/letters'), {
```
Comment (rosendi) [inline src/pages/automation/letters/content.ejs]: Move requires to top.

File: src/pages/automation/letters/create/content.ejs [hunk 8]
```diff
@@ -7,12 +7,17 @@
                 title: i18n('Название и параметры отправки'),
                 models: this.models,
                 collections: this.collections,
-                modifiers: {
-                    state: this.params.mode === 'create' ? 'active' : false
+                errorHandlers: {
+                     'write_error': function(error) {
```
Comment (rosendi) [inline src/pages/automation/letters/create/content.ejs]: Wrap it into a function

File: src/pages/automation/letters/create/content.ejs [hunk 9]
```diff
@@ -7,12 +7,17 @@
...
+                         var formEl = this.el.querySelector('.js-form');
+                         if (formEl && formEl.block) {
+                             formEl.block.showError(i18n('API_ERROR', error), 'name');
+                         }
+                     }
                 }
             }
         }, {
-            block: require('blocks/wizard/__step/templateEditContent/templateEditContent'),
+            block: require('blocks/wizard/__step/letterEditContent/letterEditContent'),
```
Comment (rosendi) [inline src/pages/automation/letters/create/content.ejs]: Require

File: src/pages/automation/letters/create/content.ejs [hunk 10]
```diff
@@ -7,12 +7,17 @@
                 title: i18n('Название и параметры отправки'),
```
Comment (rosendi) [inline src/pages/automation/letters/create/content.ejs]: Avoid using `this`

File: src/pages/automation/letters/create/content.ejs [hunk 11]
```diff
@@ -7,12 +7,17 @@
...
                 models: this.models,
                 collections: this.collections,
-                modifiers: {
-                    state: this.params.mode === 'create' ? 'active' : false
+                errorHandlers: {
```
Comment (rosendi) [inline src/pages/automation/letters/create/content.ejs]: Move to the step block.

File: src/pages/automation/letters/create/index.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+            templates: function() {
+                return new TemplateCollection({
+                    fetchModels: true
+                });
+            }
+        },
+
+        initialize: function() {
+            var _super =  Page.prototype.initialize.apply(this, arguments);
+
```
Comment (rosendi) [inline src/pages/automation/letters/create/index.js]: Remove the new line

**Comment by rosendi** [comment]
LGTM

**Pull request merged**